File: src/com/statix/android/settings/overlay/FeatureFactoryImplStatix.java
package com.statix.android.settings.overlay;

import android.content.Context;

import com.android.settings.accounts.AccountFeatureProvider;
import com.android.settings.fuelgauge.BatteryStatusFeatureProvider;
import com.android.settings.overlay.FeatureFactoryImpl;

import com.google.android.settings.accounts.AccountFeatureProviderGoogleImpl;
import com.statix.android.settings.fuelgauge.BatteryStatusFeatureProviderStatixImpl;

public final class FeatureFactoryImplStatix extends FeatureFactoryImpl {

    private AccountFeatureProvider mAccountFeatureProvider;
    private BatteryStatusFeatureProvider mBatteryStatusFeatureProvider;

    @Override
    public AccountFeatureProvider getAccountFeatureProvider() {
        mAccountFeatureProvider = new AccountFeatureProviderGoogleImpl();
        return mAccountFeatureProvider;
    }
}

File: src/com/statix/android/settings/overlay/FeatureFactoryImplStatix.kt
package com.statix.android.settings.overlay

import android.content.Context

import com.android.settings.accounts.AccountFeatureProvider
import com.android.settings.overlay.FeatureFactoryImpl
import com.android.settings.search.SearchFeatureProvider

import com.google.android.settings.accounts.AccountFeatureProviderGoogleImpl
import com.google.android.settings.search.SearchFeatureProviderGoogleImpl

class FeatureFactoryImplStatix : FeatureFactoryImpl() {
    override val accountFeatureProvider: AccountFeatureProvider by lazy {
        AccountFeatureProviderGoogleImpl()
    }

    override val searchFeatureProvider: SearchFeatureProvider = SearchFeatureProviderGoogleImpl
}
